pkg/execution/batch/batch_test.go

```diff
@@ -59,6 +59,54 @@ func TestBatchSizeLimit(t *testing.T) {
 	require.Equal(t, enums.BatchMaxSize, res.Status)
 }
 
+func TestBatchAppendIdempotence(t *testing.T) {
```

Comment:
If you're feeling spicy, maybe another test that ensures the same event can be appended to 2 different batches. Reading the `append.lua`, that shouldn't happen. But a test would be awesome documentation of that!